Paper: Visualization of Bioinformatics Data with Dash Bio

shammamah-zz opened the pull request.

Commit:
Add all sections and write introduction and abstract.

Commit:
Add descriptions of all components.

Commit:
Update abstract, introduction, and descriptions for some components.

Commit:
Add images and update descriptions.

Commit:
Add mol3d and speck.

Commit:
Add links to all external libraries used and complete sections for some more components.

Commit:
Add sequence viewer image.

Commit:
Add section on gene expression parser and FASTA parser.

Commit:
Remove XYZ reader section.

Commit:
Add PDB parser section.

Commit:
Small wording changes.

Commit:
Add more information for components; fix minor wording errors.

Commit:
Small wording changes.

Commit:
Alphabetize references.

Commit:
Standardize reference styles.

Commit:
Add all images.

deniederhut commented:
Hi @shammamah ! Your reviewer (Kirtan Dave) misunderstood the review instructions and submitted their review to me via email. I'm pasting it below, and will follow up with you myself about the suggested changes. Here it is:

> Comments:
> Visualization of Bioinformatics Data with Dash Bio
> 1. The paper contains interesting Visualization of Bioinformatics Data
> 3. There are lot of scope for improvement in all part of paper to make more scientifically sound. The sample code presented here is not provided with handling of genome file format like. .bam,.sam,.gtk etc. Also, presented visualization of chromosome is only with .json file. How user can view genome with .bam and .bai file?
> 6. How the Dash Bio incorporated with other Bioinformatics data analysis pkg ?
> 7. How the visualization from Dash Bio is different from other available in Biopython?
> 8. It is not clear whether Dash Bio is cover all Bioinformatics file format and handling all kind file format for visualization. If yes so why no example presented or if no, how future improvement can be done?

> Best
> Kirtan

shammamah-zz commented:
@deniederhut Thank you very much!

deniederhut commented on papers/shammamah_hossain/shammamah_hossain.rst:
```diff
@@ -0,0 +1,798 @@
+   >>> import json
+   >>> from dash_bio_utils import pdb_parser
+   >>> pdb_string = pdb_parser.create_data('1bna.pdb')
+   >>> 1bna = json.loads(pdb_string)
```
Was this supposed to be `lbna`? Python names can't start with digits.

shammamah-zz replied:
Good catch! Fixed in ff23d98.

deniederhut reviewed (changes requested):
Cool! Really cool! It looks like you're missing DOIs for some of the papers you cite. Would you mind adding them in?

Commit:
Fix variable name.

Commit:
Add DOI to all papers cited>

shammamah-zz commented:
@deniederhut I've added in all DOIs here: 54cd812

I'll continue to address the comments in the review that I got over the next few days. Thank you!

shammamah-zz commented:
Regarding this comment: 

>There are lot of scope for improvement in all part of paper to make more scientifically sound. 

Would it be possible to have more clarification on what exactly this means (i.e., what the paper is missing that, if added, would make it more scientifically sound)?

Commit:
Add section about future developments to the package.

Commit:
Add section with comparisons to existing tools.

deniederhut commented:
> i.e., what the paper is missing that, if added, would make it more scientifically sound

🤷‍♀️

So, the intention behind the way we do paper editing is that you and your reviewer have a constructive conversation, and arrive at a paper together that is clear and tightly constructed. Since Kirtan is not here to have a dialogue with you, I will say that the paper looked sound to me, although I am not a computational biologist. You are free to ignore that particular comment.

Please ping me when you feel the paper is ready for me to have a second look 😄

Commit:
Clarify more re: advantages of AlignmentViewer.

shammamah-zz commented:
@deniederhut I believe it should be ready for a second review now! 😄

I've addressed the comments as follows: 

>There are lot of scope for improvement in all part of paper to make more scientifically sound. 

Ignoring this, as per your suggestion. 

>The sample code presented here is not provided with handling of genome file format like. .bam,.sam,.gtk etc. Also, presented visualization of chromosome is only with .json file. How user can view genome with .bam and .bai file?

This is a limitation of the package. A section on limitations and future directions has been added here: 15610ea

>How the Dash Bio incorporated with other Bioinformatics data analysis pkg ?

Again, I'm not 100% sure what this means. There is a part in the File Parsers section that mentions `GEOparse` and `parmed`, but I'm not sure if that's enough. 

>How the visualization from Dash Bio is different from other available in Biopython?

This has been addressed in 2190e5c and ae6ccbd (but again, not sure if that's enough). 

>It is not clear whether Dash Bio is cover all Bioinformatics file format and handling all kind file format for visualization. If yes so why no example presented or if no, how future improvement can be done?

This is addressed in 15610ea.

deniederhut reviewed (approved):
Looks good to me. Thanks for submitting to SciPy!
